src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java

```diff
@@ -43,27 +31,43 @@
 import org.opensearch.search.fetch.subphase.FetchSourceContext;
 import org.opensearch.search.internal.InternalSearchResponse;
 import org.opensearch.search.profile.SearchProfileShardResults;
-import org.opensearch.search.relevance.configuration.ConfigurationUtils;
 import org.opensearch.search.relevance.client.OpenSearchClient;
+import org.opensearch.search.relevance.configuration.ConfigurationUtils;
 import org.opensearch.search.relevance.configuration.ResultTransformerConfiguration;
+import org.opensearch.search.relevance.configuration.ResultTransformerConfigurationFactory;
 import org.opensearch.search.relevance.transformer.ResultTransformer;
-import org.opensearch.search.relevance.transformer.ResultTransformerType;
-import org.opensearch.search.suggest.Suggest;
 import org.opensearch.tasks.Task;
 
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
 public class SearchActionFilter implements ActionFilter {
     private static final Logger logger = LogManager.getLogger(SearchActionFilter.class);
 
     private final int order;
 
     private final NamedWriteableRegistry namedWriteableRegistry;
-    private final Map<ResultTransformerType, ResultTransformer> supportedResultTransformers;
+    private final Map<String, ResultTransformer> resultTransformerMap = new HashMap<>();
+    private final Map<String, ResultTransformerConfigurationFactory> configurationFactoryMap = new HashMap<>();
```

Comment:
This is unused. 

I was thinking of constructing configs separate from selecting the transformers, but then I never followed through. (Instead I just added a `getConfigurationFactory()` method to the `ResultTransfomer` interface.)